### ISSUE TITLE
Better integrations

### ISSUE THREAD

**Issue by paulfitz** [gristlabs/grist-core#41]
Every app needs to integrate with every other app these days. Grist supports Zapier, has an API, and uses an open data format (Sqlite), but could do more to support the universe of possible integrations.

**Comment by anaisconce** [comment]
Note to include Google Drive syncing in the universe of better integrations.

**Comment by csdoes** [comment]
Pabbly please!

**Comment by dsagal** [comment]
We started a poll at the community forum at https://community.getgrist.com/t/what-data-view-and-3-rd-party-integrator-would-you-like-to-see-next/37 (where Pabbly is currently winning).

**Comment by paulfitz** [comment]
I've split Pabbly support out into https://github.com/gristlabs/grist-core/issues/74 - we're working on it @csdoes

**Comment by paulfitz** [comment]
Other issues relevant to better integrations are #76 (webhooks) and #75 (record-based API).

**Comment by paulfitz** [comment]
Update: Pabbly now supports Grist actions: https://www.pabbly.com/connect/integrations/grist/

**Comment by anaisconce** [comment]
Note to add support for importing sqlite files

**Comment by anaisconce** [comment]
Update: Integrately now supports Grist integrations!

**Comment by anaisconce** [comment]
Request to add IFTTT https://ifttt.com/  💪

**Comment by dsagal** [comment]
Update. Here are the links to integrators who support Grist, as of September 2021:
- Zapier: https://zapier.com/apps/grist/integrations
- Integrately: https://integrately.com/integrations/grist
- Pabbly: https://www.pabbly.com/connect/integrations/grist/
- KonnectzIT: https://plan.konnectzit.com/feedback/grist-integration

**Comment by asitemade4u** [comment]
n8n is the only open source self hosted tool in your list -- amazing piece of software.
Unfortunately it only works with the website Grist, not the self hosted one.

**Comment by paulfitz** [comment]
Good point @asitemade4u - we'll look at tweaking the n8n integration. In the meantime, looking at the code at https://github.com/n8n-io/n8n/commit/4bce33a5306585d0b767b0db8dbf0ae0ac0f8836, getgrist.com is just mentioned in two non-comments, so if you were up for it it might be an easy change to make yourself.

**Comment by tennox** [comment]
@asitemade4u @paulfitz done ;)
https://github.com/n8n-io/n8n/pull/2788

**Comment by asitemade4u** [comment]
Splendid, you're on fire!

**Comment by almereyda** [comment]
Meanwhile https://github.com/n8n-io/n8n/pull/2847 got merged. Are there already sketches about how a Grist Trigger for n8n could look like?

**Comment by paulfitz** [comment]
@almereyda the code from the Zapier integration might be a helpful reference:
  * https://github.com/gristlabs/grist-zapier
  * an example instant trigger - https://github.com/gristlabs/grist-zapier/blob/main/triggers/updated_record_instant.js
  * this relies on our as yet unofficial api for webhooks - https://github.com/gristlabs/grist-core/issues/76#issuecomment-1025802544

**Comment by almereyda** [comment]
😍 Webhooks ...  maybe some time in autumn I could try to adapt the Zapier example, if noone else steps up earlier.

**Comment by brandonkal** [comment]
Requesting integration with [make.com](https://make.com) Thanks!

**Comment by audunmb** [comment]
Adding Grist to NextCloud would be great. Currently NextCloud seems focused on Collabora/LibreOffice, but innovative solutions like Grist would be a better way forward for NextCloud. For Grist it would mean access to all the European users looking for GDPR-friendly Office 365 replacements. 

I guess the way to do this would be to add Grist as an "app" in NextCloud.

**Comment by anaisconce** [comment]
Hm, the user who created the Grist template for Unraid is also a NextCloud user. Let me ping him and see if he might have insights about how to do this, @audunmb 

@brandonkal I see Grist on make.com https://www.make.com/en/integrations/grist (formerly Integromat!)

**Comment by mr6volt** [comment]
Not sure if this is the right place to report this, but the integrations for Grist on Make.com and Zapier do not work with the Self-Hosted version of Grist.

Make.com keeps tacking on .getgrist.com to domains, and borking connections.

Zapier doesn't appear to be able to go past Team selection. (It can't find Documents) Maybe an outdated api call?

**Comment by slimhk45** [comment]
+1 for a Grist NextCloud app that can open .grist file stored on NextCloud. 

There is already an app named NextCloud Tables but it is a bad impementation. The data is not even exportable, or stored in the NextCloud storage as a file like Grist. 

https://github.com/nextcloud/tables

**Comment by ouvry-ems** [comment]
+1 for opening Grist documents in NextCloud.

But I would add the reverse way also: importing spreadsheets documents from Nextcloud. Currently there is only Google Drive plugin but we could have a Nextcloud plugin to import from Nextcloud!

**Comment by almereyda** [comment]
These Outline discussion give an idea about what the other side of an integration with Grist might look for:

- https://github.com/outline/outline/discussions/6507
- https://github.com/outline/outline/discussions/8476

*Edit:* Another thing that came to my mind after writing https://github.com/outline/outline/discussions/8476#discussioncomment-14780020: Are there plans to not only embed Grist via iframe's, but possibly by including a JavaScript HTML snippet loading a document view natively into the DOM?

- Would that allow the embeddor to access the data in the guest document?
- Would that already be possible with the iframe?

There are numerous ways to approach this, why finding and choosing appropriate use cases seems a first.
